pkg/metrics/config.go

```diff
@@ -1,11 +1,20 @@
 package metrics
 
+import (
+	"github.com/pkg/errors"
+)
+
 // PrometheusConfig defines Prometheus configuration.
 type PrometheusConfig struct {
-	Url string `yaml:"url"`
+	Url      string `yaml:"url"`
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
 }
 
 // Validate checks constraints in the supplied Prometheus configuration and returns an error if they are violated.
 func (c *PrometheusConfig) Validate() error {
+	if (c.Username == "") != (c.Password == "") {
```

Comment:
`notifications.Config` only validates configuration if `url` is set, i,e. `if c.Url != "" ...`. We could do the same here so that validation works the same for both.